weather_providers/openweathermap.py

```diff
@@ -0,0 +1,102 @@
+import logging
+import requests
+import os
+import json
+
+from utility import get_response_data, is_daytime
+
+
+# Map OpenWeatherMap icons to local icons
+# Reference: https://openweathermap.org/weather-conditions
+def get_icon_from_openweathermap_weathercode(weathercode, is_daytime):
+
+    icon_dict = { 
+            200 : "thundershower_rain",
+            201 : "thundershower_rain",
+            202 : "thundershower_rain",
+            210 : "thundershower_rain",
+            211 : "thundershower_rain",
+            212 : "thundershower_rain",
+            221 : "thundershower_rain",
+            230 : "thundershower_rain",
+            231 : "thundershower_rain",
+            232 : "thundershower_rain",
+            300 : "climacell_drizzle" if is_daytime else "rain_night",
+            301 : "climacell_drizzle" if is_daytime else "rain_night",
+            302 : "climacell_rain" if is_daytime else "rain_night",
+            310 : "climacell_drizzle" if is_daytime else "rain_night",
+            311 : "climacell_drizzle" if is_daytime else "rain_night",
+            312 : "climacell_rain" if is_daytime else "rain_night",
+            313 : "climacell_rain" if is_daytime else "rain_night",
+            314 : "climacell_rain_heavy" if is_daytime else "rain_night",
+            321 : "climacell_drizzle" if is_daytime else "rain_night",
+            500 : "climacell_rain_light" if is_daytime else "rain_night",
+            501 : "climacell_rain" if is_daytime else "rain_night",
+            502 : "climacell_rain_heavy" if is_daytime else "rain_night",
+            503 : "climacell_rain_heavy" if is_daytime else "rain_night",
+            504 : "climacell_rain_heavy" if is_daytime else "rain_night",
+            511 : "climacell_freezing_rain",
+            520 : "climacell_rain_light" if is_daytime else "rain_night",
+            521 : "climacell_rain" if is_daytime else "rain_night",
+            522 : "climacell_rain_heavy" if is_daytime else "rain_night",
+            531 : "climacell_rain" if is_daytime else "rain_night",
+            600 : "climacell_snow_light",
+            601 : "snow",
+            602 : "snow",
+            611 : "rain_snow_mix",
+            612 : "rain_snow_mix",
+            613 : "rain_snow_mix",
+            615 : "rain_snow_mix",
+            616 : "rain_snow_mix",
+            620 : "climacell_snow_light",
+            621 : "snow",
+            622 : "snow",
+            701 : "climacell_fog",
+            711 : "fire_smoke",
+            721 : "climacell_fog",
+            731 : "", # sand/dust whirls
+            741 : "climacell_fog",
+            751 : "", # sand
+            761 : "", #dust
+            762 : "", #ash
```

Comment:
Can't find any icons for these.  It'll just have to be blank I guess.

Reply:
I found some and applied them.  Volcano looks interesting but I hope nobody ever sees it on their dashboard!